src/Graphics/Texture2D.cs

```diff
@@ -179,7 +187,15 @@ int elementCount
 				w = Math.Max(Width >> level, 1);
 				h = Math.Max(Height >> level, 1);
 			}
-			int elementSize = Marshal.SizeOf(typeof(T));
+			int elementSize = Marshal.SizeOf(typeof(T)),
+				blockSize = GetBlockSize(Format),
```

Comment:
Do we want to cheat and make this GetBlockSizeSquared?

Reply:
I'm fine with either